### PR TITLE
Fix build instructions regarding how to debug optee-os and TA's in qemu

### PR DESCRIPTION
**Pull request by Erez-Shlingbaum**
ASLR has been enabled by default some time ago and the docs were not updated.

Signed-off-by: Erez-Shlingbaum <erez.shlingbaum@gmail.com>

### CODE REVIEW

**Comment by jforissier** [comment]
Thanks for the patch, it is good advice to give for general troubleshooting.

`Reviewed-by: Jerome Forissier <jerome@forissier.org>`

Please note that symbolic debugging with ASLR enabled is still possible, and actually not that hard, although slightly inconvenient. FYI, here are the notes I have in my ~/.gdbinit:

```
define optee_v8
        symbol-file ~/work/optee_repo_qemu_v8/optee_os/out/arm/core/tee.elf
        # Note: to debug tee.elf with CFG_CORE_ASLR:
        # 1. Use the normal symbol-file command above and break after init_mem_map()
        #    For example: b check_mem_map
        # 2. Check for debug message: "Mapping core at 0xbadab000" then do:
        # add-symbol-file ~/work/optee_repo_qemu_v8/optee_os/out/arm/core/tee.elf 0xbadab000

        # To debug a TA
        # Note: Use load address given by OP-TEE + 0x20 (= start of .text; first PT_LOAD is .ta_header)
        # Note: With TA ASLR this address will change, without it (CFG_TA_ASLR=n) it will likely remain constant
        #add-symbol-file ~/work/optee_repo_qemu_v8/out-br/build/optee_test_ext-1.0/ta/os_test/out/5b9e0e40-2636-11e1-ad9e-0002a5d5c51b.elf 0x40014020

        target remote localhost:1234
end
document optee_v8
        Loads and setup the binary (tee.elf) for OP-TEE and also connects to the QEMU
        remote.
end
```

Perhaps I should take some time to write this properly into the doc.

**Comment by Erez-Shlingbaum** [comment]
Thanks for the quick reply! It is indeed helpful info and I am sure it could help if you will add it to the docs.

File: building/devices/qemu.rst [hunk 1]
```diff
@@ -23,6 +23,7 @@ for Armv7-A is like the one below:
   $ repo sync
   $ cd build
   $ make toolchains
+  # Note that if you wish to debug optee-os or a TA, you should disable ASLR with flag "CFG_CORE_ASLR=n"
```
Comment (jbech-linaro) [inline building/devices/qemu.rst:26]: We've tried to break at 80 chars everywhere else in the documentation, so please continue with that pattern.

File: building/devices/qemu.rst [hunk 2]
```diff
@@ -129,9 +130,9 @@ GDB - Secure world
 ******************
 TEE core debugging
 ==================
-To debug TEE core running QEMU with GDB, you don't have to enable any special
-flags as such, but it's easier to debug if you have optimization disabled. Other
-than that you will have four consoles that you are working with.
+To debug TEE core running QEMU with GDB, you need to disable TEE ASLR with "CFG_CORE_ASLR=n" flag.
```
Comment (jbech-linaro) [inline building/devices/qemu.rst:133]: 80 char and 
```suggestion
To debug TEE core running QEMU with GDB, you need to disable TEE ASLR with ``CFG_CORE_ASLR=n`` flag.
```

File: building/devices/qemu.rst [hunk 3]
```diff
@@ -129,9 +130,9 @@ GDB - Secure world
...
+Furthermore, note that it's easier to debug if you have optimization disabled. Other than that you
```
Comment (jbech-linaro) [inline building/devices/qemu.rst:134]: 80 char.

File: building/devices/qemu.rst [hunk 4]
```diff
@@ -237,6 +238,7 @@ for Armv8-A is like the one below:
   $ repo sync
   $ cd build
   $ make toolchains
+  # Note that if you wish to debug optee-os or a TA, you should disable ASLR with flag "CFG_CORE_ASLR=n"
```
Comment (jbech-linaro) [inline building/devices/qemu.rst:241]: 80 char

**Review by jbech-linaro — COMMENTED** [review]
Thanks for the contribution, only a few minor things. In addition, you might find [my presentation](https://docs.google.com/presentation/d/1NEBZPrYqnjV55NOkJp_XQ7nGlOwDRuFb7z6cGO0Iboo/edit#slide=id.g7ddd37b719_1_67) from late last year useful as well.

Once the minor issues fixed:
`Reviewed-by: Joakim Bech <joakim.bech@linaro.org>`

**Comment by jbech-linaro** [comment]
Thanks @Erez-Shlingbaum , would you mind squash the patches, add my reviewed-by tag and force push? Once done I'll merge.

**Comment by Erez-Shlingbaum** [comment]
@jbech-linaro Done.

**Comment by jbech-linaro** [comment]
Thanks @Erez-Shlingbaum , I've tweaked the commit message slightly and then I've [pushed the patch](https://github.com/OP-TEE/optee_docs/commit/7c4e017a48941fb9475179485ba3533ab18b0c9a), therefore I'm closing the this PR.

**Pull request closed without merging**